docs/modules/ROOT/pages/explanations/decisions/postgres-upgrades.adoc

```diff
@@ -0,0 +1,83 @@
+= VSHN Managed PostgreSQL
+
+== Problem
+
+We need to specify how VSHN handles the major and minor upgrades of PostgreSQL by VSHN.
+This decision is not about upgrading StackGres, while related to this topic, it doesn't have any customer impact.
+
+So we need to decide two things:
+
+* How upgrades are handled
+* How we communicate with the customer about it
+
+== Requirements
+
+* Specification how we do upgrades
+* Specification when we do upgrades
+* Specification how customers should be informed
+* Specification when customers should be informed
+
+== Proposals
+
+=== Upgrades
+
+Proposal 1::
+Automatic Minor and major upgrades
+
+Proposal 2::
+Automatic minor upgrade and manual major upgrades
+
+Proposal 3::
+Manual minor and major upgrades
+
+=== Communication
+
+Proposal 4::
+Inform 3 months after EOL and then stop the instance
+
+Proposal 5::
+Inform 3 months after EOL and then keep the instances but unsupported
+
+Proposal 6::
+Inform 3 months after EOL and then delete the instance with enabled deletion protection
+
+Proposal 7::
+Inform 3 months after EOL and then force upgrade the instance
```

Comment:
rather before EOL, or `after EOL announcement`? 

I'd say 3 months `after EOL announcement` because it may vary between postgresql itself and Stackgres, so if stackgres announce that they won't support 12.x anymore, then we give our customers 3 months to migrate to 13.x or newer, otherwise they will be forcefully updated.